include/ItemTypes.h

```diff
@@ -2,12 +2,24 @@
 #ifndef ITEMTYPES_H_INCLUDED
 #define ITEMTYPES_H_INCLUDED
 
+#include <unordered_map>
+
 enum ITEMTYPE
 {
 	ARROWS = 1,
 	BOMB,
 	POTION,
-	WHETSTONE
+	WHETSTONE,
+	COIN
+};
+
+struct ItemTypeHash {
+    template<typename T>
+    std::size_t operator()(T t) const {
+        return static_cast<std::size_t>(t);
+    }
 };
 
+using ItemMap_t = std::unordered_map<ITEMTYPE, int, ItemTypeHash>;
```

Comment:
Hmm, can I ask why we need the `ItemTypeHash`? If it's because `std::unordered_map` requires it, then I don't think it's worth the extra complexity over `std::map`. After all, the inventory is small enough that O(logn) isn't measurably better than O(1).